wasm_for_tests/wasm_source/src/lib.rs

```diff
@@ -37,12 +37,41 @@ pub mod main {
     }
 }
 
+/// A tx that endlessly charges gas from the host environment
+#[cfg(feature = "tx_infinite_host_gas")]
+pub mod main {
+    use namada_tx_prelude::*;
+
+    #[transaction]
+    fn apply_tx(ctx: &mut Ctx, _tx_data: Tx) -> TxResult {
+        let target_key = parameters_storage::get_tx_allowlist_storage_key();
+        loop {
+            // NOTE: don't propagate the error to verify that execution abortion is done in host and does not require guest cooperation
+            let _ = ctx.write(&target_key, vec!["hash"]);
+        }
+    }
+}
+
+/// A tx that endlessly charges gas from the guest environment
+#[cfg(feature = "tx_infinite_guest_gas")]
+pub mod main {
+    use namada_tx_prelude::*;
+
+    #[transaction]
+    fn apply_tx(_ctx: &mut Ctx, _tx_data: Tx) -> TxResult {
+        let mut _a = 0u64;
+        loop {
+            _a += 1;
+        }
```

Comment:
does execution halt if we only have an empty loop, like:
```rs
loop {}
```
??

Reply:
Ah nice one. We do charge gas for branch instructions too so I'd say yes, but let me try to rework this tx (and the vp one) and see if the tests keep working even with empty loops

Reply:
Ok it seems to be working even with an empty loop, I've pushed the updated test wasms